src/respend/respenddetector.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "respend/respenddetector.h"
+#include "bloom.h"
+#include "respend/respendaction.h"
+#include "respend/respendlogger.h"
+#include "respend/respendrelayer.h"
+#include "txmempool.h"
+#include "util.h"
+
+#include <algorithm>
+
+namespace respend
+{
+static const unsigned int MAX_RESPEND_BLOOM = 100000;
+std::unique_ptr<CRollingBloomFilter> RespendDetector::respentBefore;
+std::mutex RespendDetector::respentBeforeMutex;
+
+std::vector<RespendActionPtr> CreateDefaultActions()
+{
+    std::vector<RespendActionPtr> actions;
+
+    actions.push_back(RespendActionPtr(new RespendRelayer{}));
+    if (LogAcceptCategory(Logging::RESPEND))
+    {
+        actions.push_back(RespendActionPtr(new RespendLogger{}));
+    }
+    return actions;
+}
+
+RespendDetector::RespendDetector(const CTxMemPool &pool, const CTransaction &tx, std::vector<RespendActionPtr> actions)
+    : actions(actions)
+{
```

Comment:
a bit of shadowing here.

-RespendDetector::RespendDetector(const CTxMemPool &pool, const CTransaction &tx, std::vector<RespendActionPtr> actions)
    : actions(actions)
+RespendDetector::RespendDetector(const CTxMemPool &pool, const CTransaction &tx, std::vector<RespendActionPtr> _actions)
    : actions(_actions)